plugin/backend.go

```diff
@@ -58,94 +65,167 @@ func Backend() *GcpAuthBackend {
 			},
 			pathsRole(b),
 		),
+
+		Invalidate: b.invalidate,
 	}
 	return b
 }
 
-func (b *GcpAuthBackend) httpClient(ctx context.Context, s logical.Storage) (*http.Client, error) {
-	config, err := b.config(ctx, s)
+// IAMClient returns a new IAM client. The client is cached.
+func (b *GcpAuthBackend) IAMClient(s logical.Storage) (*iam.Service, error) {
+	httpClient, err := b.httpClient(s)
```

Comment:
Could this be moved inside the fetch closure? (and in the other functions)

Reply:
@briankassouf it cannot because the HTTP client itself is also inside a fetch closure, which would create a deadlock. The first invocation of `b.httpClient` will create the cache.